File: sdi3-14.Cli-REST/src/uo/sdi/clients/rest/menu/items/ListCategoriesMenuItem.java
package uo.sdi.clients.rest.menu.items;

import java.util.List;

import uo.sdi.clients.rest.dto.Category;
import uo.sdi.clients.rest.restService.RestService;
import alb.util.console.Console;

public class ListCategoriesMenuItem extends MainMenuItem {

	public ListCategoriesMenuItem(RestService restService) {
		super(restService);
	}

	@Override
	public String getMenuName() {
		return "Ver categorias";
	}

	@Override
	protected void executeInternal() {

		List<Category> categories = restService.findCategoriesByUserID();

		for (Category c : categories) {
			Console.println(c);
		}

	}

}

File: sdi3-14.Cli-REST-CSharp/sdi3-14.Cli-REST-CSharp/menu/menuItems/ListCategoriesMenuItem.cs
using sdi3_14.Cli_REST_CSharp.dto;
using sdi3_14.Cli_REST_CSharp.restService;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_14.Cli_REST_CSharp.menu.menuItems
{
    class ListCategoriesMenuItem : MainMenuItem
    {

        public ListCategoriesMenuItem(RestService restService) : base(restService)
        {
        }

        public override String GetMenuName()
        {
            return "Ver categorias";
        }

        protected override void executeInternal()
        {

            List<Category> categories = restService.findCategoriesByUserID();

            foreach (Category c in categories)
            {
                System.Console.WriteLine(c);
            }

        }

    }
}
